.github/build.yaml.gomplate

```diff
@@ -86,8 +86,9 @@
           timeout_minutes: 5
           max_attempts: 3
           command: |
-            curl -L https://golang.org/dl/go1.17.5.linux-{{{ $arch }}}.tar.gz -o go1.17.5.linux-{{{ $arch }}}.tar.gz
-            sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.17.5.linux-{{{ $arch }}}.tar.gz
+            curl -L https://golang.org/dl/go1.18.8.linux-{{{ $arch }}}.tar.gz -o go1.18.8.linux-{{{ $arch }}}.tar.gz
```

Comment:
we should probably create a template that returns the go version so we can use it everywhere and we we want to bump up the go version we can do in a single place before running the pipelines.sh otherwise we gonna keep changing the version manually :/

Reply:
Yes I also thought about that, but first I wanted to check if my approach is actually working. So far I already found an inconsistency at `.github/workflows/build-main-teal-arm64.yaml`, as it runs on ubuntu-latest but it renders the steps for self hosted runners... the build gomplate is not well prepared to distinguish this sort of cases were we run some cross-arch stuff... So arm with this PR is definitely broken.

Reply:
Ok, now I think it should be fine. I'd merge this and check if the publish job is now fixed. If so a follow up PR to template the golang version should be easy and with little chances to break again something.